includes/process-purchase.php

```diff
@@ -74,6 +74,31 @@ function edd_process_purchase_form() {
 		'address'    => $user['address']
 	);
 
+	// Update a customer record if they have added/updated information
+	if ( is_user_logged_in() ) {
```

Comment:
This shouldn't require the customer be logged in. As soon as we have a customer record identified, we can update it, logged in or logged out.